src/mpi/coll/ireduce/ireduce.c

```diff
@@ -20,6 +20,18 @@
       description : >-
         k value for tree based ireduce (for tree_kary and tree_knomial)
 
+    - name        : MPIR_CVAR_IREDUCE_TREE_PIPELINE_CHUNK_SIZE
```

Comment:
This has nothing to do with the ring algorithm, right?

Reply:
In addition to ring based algo for reduce, this PR has two additional commits that enable pipelining in the already existing tree algo for reduce. See first comment in this PR. I should update the PR title.

Reply:
For completeness, in addition, it has one more commit - `Use separate tags for each task`. This is the only PR with more than 1 subtopic commit, PRs after this are consistent.